symengine/add.h

```diff
@@ -69,11 +69,11 @@ class Add : public Basic
 
     virtual vec_basic get_args() const;
 
-    RCP<const Number> get_coef() const
+    inline const RCP<const Number> &get_coef() const
     {
         return coef_;
     }
```

Comment:
Ok, so the get_coef() is being passed around. In that case, one can't return `&`, because I think the RCP might not get activated properly.

The solution for things where the result is not passed around is to return a `Ptr`:
```diff
-    inline const RCP<const Number> &get_coef() const
+    inline const Ptr<const Number> get_coef() const
     {
-        return coef_;
+        return coef_.ptr();
     }
```
That is safe.